test/ring/middleware/oauth2_test.clj

```diff
@@ -56,6 +56,15 @@
         location  (get-in response [:headers "Location"])]
     (is (.startsWith ^String location "https://example.com/oauth2/authorize?business_partner_id=XXXX&"))))
 
+(deftest test-location-uri-with-dynamic-query
+  (let [profile   test-profile
+        handler   (wrap-oauth2 token-handler {:test profile})
+        response  (handler (-> (mock/request :post "/oauth2/test?hd=tenant.com" {:hd "tenant.com"})
```

Comment:
ah this is a little weird. was having trouble finding these params in the handler, but added that `wrap-params` middleware and committed once the tests passed.

should this be 

```suggestion
        response  (handler (-> (mock/request :post "/oauth2/test" {:hd "tenant.com"})
```